### ISSUE TITLE
.NET MAUI support

### ISSUE THREAD

**Issue by vhugogarcia** [microsoft/appcenter#2268]
**Describe the solution you'd like**
A library compatible with MAUI. 

**Additional context**
Next month, we will start a new project on .NET MAUI using the RC1. So, we would like to know if app center will provide a nuget package compatible with .NET MAUI. 

Thank you

**Comment by DmitriyKirakosyan** [comment]
Hi @vhugogarcia 

Thank you for reaching out to us!

We will test AppCenter SDK with MAUI and let you know if it works as expected.

**Comment by vhugogarcia** [comment]
Thank you so much! Really appreciate your support 🙂

**Comment by timvandenhof** [comment]
Will follow this as well while we are setting up plans on migrating from Xamarin.Forms to MAUI. We sure would like to keep using App Center (Diagnostics, Test, Internal Android distribution). Seeing the Roadmap post about removing techdebt, I hope they are working on some technical revamp to make this all work.

**Comment by DmitriyKirakosyan** [comment]
Hi guys,

I checked the App Center SDK compatibility with MAUI and everything looked good. SDK was successfully integrated and worked as expected on Android and WinUI apps.
I haven't checked iOS as I didn't set up my windows -> mac, but since Android works, there is no reason for iOS to have any issues.

Note, that I tested internal NuGets from the `developer` branch. NuGets from official release don't support WinUI yet. We will release a new SDK version this month.

App Center CI/CD cervices including Build, Test and Distribution do not support MAUI yet. Please file a separate feature request if you are interested in these services as well.

So answering on the original question, Yes, SDK NuGet package support MAUI for iOS and Android, and will support WinUI starting from the next release.
I'm closing this issue, but feel free to reopen if you have additional questions.

**Comment by vhugogarcia** [comment]
Thanks @DmitriyKirakosyan for the detailed response. 👍🏻  That helped a lot.

**Comment by IndexOutOfLimit** [comment]
Is there any release available supporting .Net6 and Maui

**Comment by DmitriyKirakosyan** [comment]
Hi @IndexOutOfLimit , yes we support .Net 6 and Maui in the current release. Feel free to open a new ticket in [appcenter-sdk-dotnet](https://github.com/microsoft/appcenter-sdk-dotnet) repo if you face any issue integrating AppCenter SDK in your app.

**Comment by IndexOutOfLimit** [comment]
Hi Dima,

When I added AppCenter nuget (v 4.4.0) in the Maui solution it is
throwing an error because it doesn't have support for net6.

**Comment by AnastasiaKubova** [comment]
Hi, @IndexOutOfLimit! Could you please the full error message? Which platforms do you use in the MAUI project?

**Comment by IndexOutOfLimit** [comment]
I am targeting Windows platform in MAUI and after adding the nuget I got many errors attached here. 

[AppCenterErrors.txt](https://github.com/microsoft/appcenter/files/7661132/AppCenterErrors.txt)

**Comment by AnastasiaKubova** [comment]
@IndexOutOfLimit Thanks for the details. Your bug probably related to this [issue](https://github.com/microsoft/appcenter-sdk-dotnet/issues/1582). We already prepared a fix for it. 
I will notify you when the fix will be available.

**Comment by IndexOutOfLimit** [comment]
@AnastasiaKubova, Is it fixed now?

**Comment by AnastasiaKubova** [comment]
Hi, @IndexOutOfLimit ! Yes, the [fix](https://github.com/microsoft/appcenter-sdk-dotnet/releases/tag/4.5.0) has been released.

**Comment by IndexOutOfLimit** [comment]
Thanks @AnastasiaKubova for your quick response 👍🏻

**Comment by carmanj** [comment]
We're waiting on this too. We're actively working on migrating from Xarmain.IOS to a MAUI-based app. It would be super helpful to us if we could continue to use AppCenter (which we currently use for CD/CI/build/distribution/diagnostics) for the MAUI app. I understand that MAUI is still in preview, but if AppCenter could also support it in a 'preview' mode, that would be greatly beneficial to us. This would avoid us having to manually provision/distribute devices to our internal Alpha testers and other stakeholders.

Thanks!

**Comment by carmanj** [comment]
> Hi guys,
> 
> I checked the App Center SDK compatibility with MAUI and everything looked good. SDK was successfully integrated and worked as expected on Android and WinUI apps. I haven't checked iOS as I didn't set up my windows -> mac, but since Android works, there is no reason for iOS to have any issues.
> 
> Note, that I tested internal NuGets from the `develop` branch. NuGets from official release don't support WinUI yet. We will release a new SDK version this month.
> 
> App Center Portal CI/CD services including Build, Test and Distribution do not support MAUI yet. However. it could work for iOS and Anroid if you create a Xamarin.iOS or Xamarin.Android apps. Please file a separate feature request if you are interested in these services as well.
> 
> So answering on the original question, Yes, SDK NuGet packages support MAUI for iOS and Android, and will support WinUI starting from the next release. I'm closing this issue, but feel free to reopen if you have additional questions.

We're waiting on this too. We're actively working on migrating from Xarmain.IOS to a MAUI-based app. It would be super helpful to us if we could continue to use AppCenter (which we currently use for CD/CI/build/distribution/diagnostics) for the MAUI app. I understand that MAUI is still in preview, but if AppCenter could also support it in a 'preview' mode, that would be greatly beneficial to us. This would avoid us having to manually provision/distribute devices to our internal Alpha testers and other stakeholders.

**Comment by DmitriyKirakosyan** [comment]
Hi @carmanj , thank you for reaching out! App Center SDK currently works on MAUI for iOS, Android and WinUI. However, as you mentioned, it is in preview, so anything may happen.
Please let us know if you faced any issue integrating App Center SDK into MAUI app.

**Comment by MichaelShapiro** [comment]
Please add support for MAUI projects.

**Comment by DmitriyKirakosyan** [comment]
Hi @MichaelShapiro , we are currently working on it. You can fallow the progress in https://github.com/microsoft/appcenter-sdk-dotnet/issues/1647.

**Comment by stephenhauck** [comment]
If Appcenter crashlytics works in MAUI apps what is the recommended initialization code?

**Comment by DmitriyKirakosyan** [comment]
@stephenhauck  AppCenter Crashes SDK doesn't rely on the app lifecylce. I would recommend to initialize it before any potential crash can happen.

**Comment by stephenhauck** [comment]
Sure it does .. if you use Xamarin Forms it's initialized in App.xaml.cs where the lifecycle events happen 
IN MAUI they don't have the lifecycle events in mauiprogram.cs but seem to encourage you to map to them with platform specific implementations through the use of ConfigureLifecycleEvents ?

**Comment by DmitriyKirakosyan** [comment]
@stephenhauck , Since MAUI App class does not have lifecycle callbacks, I believe you can call AppCenter.Start right in the constructor after all default initializations. No need to attach the sdk initialization to a platform specific lifecycle. The only recommendation is that you do it in the main thread to avoid any issues with crashes happening at the same time from another thread.

Note, that AppCenter SDK uses lifecycle events only for session tracking.

Please let me know if initialization in ctor doesn't work for you.

**Comment by stephenhauck** [comment]
So someone at Microsoft need to document this ... 
Appcenter has been abandon it seems ... still can't build C#9 and source geenerators in Appcenter 
Until the documentation has been changed it doesn't support MAUI ... 
![image](https://user-images.githubusercontent.com/12648793/189897545-0e9abd0f-8627-48a5-a2b0-f109b14584a2.png)

**Comment by AnatolyPristensky** [comment]
@stephenhauck, initial issue regarding App Center MAUI Build is here: https://github.com/microsoft/appcenter/issues/2360
You can bump the initial issue if you mean AppCenter MAUI Build.
